### PR TITLE
[KOB-9634] Added UIAutomator samples with multiple languages

### PR DESCRIPTION
**Pull request by ngantlnguyen**

### CODE REVIEW

File: uiautomator-espresso/ruby/uiautomator.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
+require 'uri'
+require 'net/http'
+require 'json'
+require 'base64'
+
+username = ''
+apiKey = ''
+encodeAuth   = 'Basic ' + Base64.encode64(username + ':' + apiKey)
```
Comment (nghiadhd) [inline uiautomator-espresso/ruby/uiautomator.rb:8]: 4 spances

File: uiautomator-espresso/ruby/uiautomator.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+https = Net::HTTP.new(url.host, url.port);
+https.use_ssl = true
+
+request = Net::HTTP::Post.new(url)
+request['Content-Type'] = 'application/json'
+request['Authorization'] = encodeAuth
+request['Accept'] = 'application/json'
+
+configuration = {
+    'configuration': {
```
Comment (nghiadhd) [inline uiautomator-espresso/ruby/uiautomator.rb:21]: 4 spaces

File: uiautomator-espresso/ruby/uiautomator.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+require 'json'
+require 'base64'
+
+username = ''
+apiKey = ''
+encodeAuth   = 'Basic ' + Base64.encode64(username + ':' + apiKey)
+
+url = URI('https://api.kobiton.com/hub/session')
+
+https = Net::HTTP.new(url.host, url.port);
```
Comment (khanhdodang) [inline uiautomator-espresso/ruby/uiautomator.rb:12]: Remove `;`

File: uiautomator-espresso/ruby/espresso.rb [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+require 'json'
+require 'base64'
+
+username = ''
+apiKey = ''
+encodeAuth   = 'Basic ' + Base64.encode64(username + ':' + apiKey)
+
+url = URI('https://api.kobiton.com/hub/session')
+
+https = Net::HTTP.new(url.host, url.port);
```
Comment (khanhdodang) [inline uiautomator-espresso/ruby/espresso.rb:12]: Remove `;`

File: uiautomator-espresso/ruby/README.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+```ruby
+username = ''
+apiKey = ''
+```
+- Update the value for `configuration` to indicate your exptected testing device.
+
+```ruby
+{
+  'configuration': {
+        'sessionName': 'Automation test session',
```
Comment (khanhdodang) [inline uiautomator-espresso/ruby/README.md:69]: Change to 2 spaces

File: uiautomator-espresso/python/uiautomator.py [hunk 6]
```diff
@@ -0,0 +1,42 @@
+import requests
+import json
+import base64
+
+username = ''
+apiKey = ''
+
+encodeAuth = 'Basic ' + (base64.b64encode((username + ':' + apiKey).encode('utf-8'))).decode('utf-8')
```
Comment (khanhdodang) [inline uiautomator-espresso/python/uiautomator.py:8]: Why do we use encode and decode at the same time?

File: uiautomator-espresso/python/README.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+
+```bash
+brew install python
+```
+
+- To check if you have [Python](https://www.python.org/) installed, run this command in your terminal:
+
+```bash
+python --version
+python3 --version
```
Comment (khanhdodang) [inline uiautomator-espresso/python/README.md:32]: We should choose python or python3 to run the tests.

File: uiautomator-espresso/php/uiautomator.php [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+        'tests'               => array(
+          'com.example.android.testing.uiautomator.BasicSample.test',
+          'ChangeTextBehaviorTest',
+          'ChangeTextBehaviorTest#testChangeText_sameActivity'
+        )
+    )),
+  CURLOPT_POSTFIELDS => json_encode($configuration),
+  CURLOPT_HTTPHEADER => array(
+    'Content-Type: application/json',
+    'Authorization: ' .$encodeAuth,
```
Comment (khanhdodang) [inline uiautomator-espresso/php/uiautomator.php:42]: `.$encodeAuth` we should change to `($encodeAuth)`

File: uiautomator-espresso/php/uiautomator.php [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+
+$username = '';
+$apiKey = '';
+
+$encodeAuth = 'Basic ' .base64_encode($username. ':' .$apiKey);
+
+curl_setopt_array($curl, array(
+  CURLOPT_URL => 'https://api.kobiton.com/hub/session',
+  CURLOPT_RETURNTRANSFER => true,
+  CURLOPT_ENCODING => '',
```
Comment (khanhdodang) [inline uiautomator-espresso/php/uiautomator.php:12]: Should remove this

File: uiautomator-espresso/php/espresso.php [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+
+$username = '';
+$apiKey = '';
+
+$encodeAuth = 'Basic ' .base64_encode($username. ':' .$apiKey);
+
+curl_setopt_array($curl, array(
+  CURLOPT_URL => 'https://api.kobiton.com/hub/session',
+  CURLOPT_RETURNTRANSFER => true,
+  CURLOPT_ENCODING => '',
```
Comment (khanhdodang) [inline uiautomator-espresso/php/espresso.php:12]: Remove this one

File: uiautomator-espresso/python/uiautomator.py (hunk 6 above)
Comment (ngantlnguyen) [reply uiautomator-espresso/python/uiautomator.py:8]: Python will throw "b 'bmdhbnRsbmd1eWVudGVzdDo0MjMxNThjZS05OTMzLTRkZDMtYjJjZS1kMjhjMWNiNzY2MTk='" (binary) if we do not use decode, and the request header will be incorrect.

File: uiautomator-espresso/python/README.md (hunk 7 above)
Comment (ngantlnguyen) [reply uiautomator-espresso/python/README.md:32]: I want to show that we can run this script with both python & python3.
Consider removing python & keep python3.

File: uiautomator-espresso/php/uiautomator.php (hunk 8 above)
Comment (ngantlnguyen) [reply uiautomator-espresso/php/uiautomator.php:42]: I used "." to concatenate the string header.

File: uiautomator-espresso/java/README.md [hunk 11]
```diff
@@ -0,0 +1,73 @@
+# Java samples using UIAutomator/ Espresso framework
+
+## I. Install the uiautomator-espresso samples
```
Comment (khanhdodang) [inline uiautomator-espresso/java/README.md:3]: Install  -> Clone

File: uiautomator-espresso/java/README.md [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+
+- Clone this script directly from our GitHub repo:
+
+```
+mkdir kobiton-samples
+cd kobiton-samples
+git clone https://github.com/kobiton/samples.git
+cd samples/uiautomator-espresso/java/
+```
+- Install [Maven](https://maven.apache.org/install.html)
```
Comment (khanhdodang) [inline uiautomator-espresso/java/README.md:13]: Remove this step

File: uiautomator-espresso/java/README.md [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+- Build project with [Maven](https://maven.apache.org/run-maven/) `mvn clean install`
```
Comment (khanhdodang) [inline uiautomator-espresso/java/README.md:14]: Remove this step

File: uiautomator-espresso/java/README.md [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+
+- Go to [Kobiton Devices page](https://portal.kobiton.com/devices).
+- Hover over the device you want to test, click on the More icon then select Automation Settings.
+
+![automation-settings.png](/uiautomator-espresso/assets/automation-settings.png)
+
+- Select Framework = UIAutomator/ Espresso.
+- Select Language = Java.
+
+![automation-settings-java.png](/uiautomator-espresso/assets/automation-settings-java.png)
```
Comment (khanhdodang) [inline uiautomator-espresso/java/README.md:27]: Check screenshot

File: uiautomator-espresso/java/README.md [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+
+- Input the value for `username` & `apiKey` & in the sample script.
+
+```
+String username = "";
+String apiKey = "";
+```
+- Update the value for `configuration` to indicate your exptected testing device.
```
Comment (khanhdodang) [inline uiautomator-espresso/java/README.md:35]: `configuration` -> `jsonObject`

File: uiautomator-espresso/javascript/README.md [hunk 16]
```diff
@@ -0,0 +1,110 @@
...
+npm -v
+```
+
+- To update your npm to latest, type this into in your terminal:
+
+```bash
+npm install npm@latest -g
+```
+
+## Install the uiautomator-espresso samples
```
Comment (khanhdodang) [inline uiautomator-espresso/javascript/README.md:41]: Install -> Clone

File: uiautomator-espresso/javascript/espresso.js [hunk 17]
```diff
@@ -0,0 +1,46 @@
+const request = require('request')
+
+const username = ''
+const apiKey = ''
+
+const encodeAuth = 'Basic ' + Buffer.from(`${username}:${apiKey}`).toString('base64')
```
Comment (khanhdodang) [inline uiautomator-espresso/javascript/espresso.js:6]: `Basic ${Buffer.from(`${username}:${apiKey}`).toString('base64')}`

File: uiautomator-espresso/php/espresso.php [hunk 18]
```diff
@@ -0,0 +1,48 @@
...
+  CURLOPT_HTTP_VERSION => CURL_HTTP_VERSION_1_1,
+  CURLOPT_CUSTOMREQUEST => 'POST',
+  $configuration  = array(
+    'configuration' => array(
+      'sessionName'         => 'Automation test session',
+      'sessionDescription'  => 'This is an example for Espresso testing',
+      'noReset'             => true,
+      'fullReset'           => false,
+      'deviceName'          => '*',
+      'platformVersion'     => '*',
```
Comment (khanhdodang) [inline uiautomator-espresso/php/espresso.php:24]: Remove platformVersion

File: uiautomator-espresso/php/espresso.php [hunk 19]
```diff
@@ -0,0 +1,48 @@
...
+  CURLOPT_MAXREDIRS => 10,
+  CURLOPT_TIMEOUT => 0,
+  CURLOPT_FOLLOWLOCATION => true,
+  CURLOPT_HTTP_VERSION => CURL_HTTP_VERSION_1_1,
+  CURLOPT_CUSTOMREQUEST => 'POST',
+  $configuration  = array(
+    'configuration' => array(
+      'sessionName'         => 'Automation test session',
+      'sessionDescription'  => 'This is an example for Espresso testing',
+      'noReset'             => true,
```
Comment (khanhdodang) [inline uiautomator-espresso/php/espresso.php:21]: Remove noReset & fullReset

File: uiautomator-espresso/javascript/README.md (hunk 16 above)
Comment (ngantlnguyen) [reply uiautomator-espresso/javascript/README.md:41]: Keep install, with javascript we need to install the dependency (request).

**Pull request merged**